### PR TITLE
[Dubbo-1681] add test for dubbo common module

### LINKED ISSUE #1681: Enhance the test coverage part-3 : dubbo-common/src/main/java/com/alibaba/dubbo/common/concurrent(io|json|logger) modules
Add unit test,modules:
  dubbo-common/src/main/java/com/alibaba/dubbo/common/concurrent
  dubbo-common/src/main/java/com/alibaba/dubbo/common/io
  dubbo-common/src/main/java/com/alibaba/dubbo/common/json
  dubbo-common/src/main/java/com/alibaba/dubbo/common/logger

### PR DESCRIPTION
**Pull request by htynkn**
## What is the purpose of the change

Improve unit test for common module.
```
dubbo-common/src/main/java/com/alibaba/dubbo/common/concurrent
dubbo-common/src/main/java/com/alibaba/dubbo/common/io
dubbo-common/src/main/java/com/alibaba/dubbo/common/json
dubbo-common/src/main/java/com/alibaba/dubbo/common/logger
```
Close #1681 
## Brief changelog

+ Add test for concurrent, io and logger.
+ Json module is deprecate since last year, ignore it in codecov config

## Verifying this change
CI Pass https://travis-ci.org/diamondblack/incubator-dubbo/builds/372700464

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/dubbo/tree/master/dubbo-test).
- [x] Run `mvn clean install -DskipTests` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [x] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/test/java/com/alibaba/dubbo/common/concurrent/ExecutionListTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+package com.alibaba.dubbo.common.concurrent;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executor;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
```
Comment (beiwei30) [inline dubbo-common/src/test/java/com/alibaba/dubbo/common/concurrent/ExecutionListTest.java]: pls. don't use * in import

File: dubbo-common/src/test/java/com/alibaba/dubbo/common/io/StreamUtilsTest.java [hunk 2]
```diff
@@ -18,11 +18,12 @@
...
 import org.junit.Test;
 
+import java.io.IOException;
 import java.io.InputStream;
 import java.io.PushbackInputStream;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.*;
```
Comment (beiwei30) [inline dubbo-common/src/test/java/com/alibaba/dubbo/common/io/StreamUtilsTest.java]: don't use '*'

File: dubbo-common/src/test/java/com/alibaba/dubbo/common/io/UnsafeByteArrayOutputStreamTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (beiwei30) [inline dubbo-common/src/test/java/com/alibaba/dubbo/common/io/UnsafeByteArrayOutputStreamTest.java]: again, no '*' but explicit import

File: dubbo-common/src/test/java/com/alibaba/dubbo/common/logger/slf4j/Slf4jLoggerTest.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alibaba.dubbo.common.logger.slf4j;
+
+import org.junit.Test;
+import org.slf4j.Marker;
+import org.slf4j.spi.LocationAwareLogger;
+
+import static org.mockito.ArgumentMatchers.*;
```
Comment (beiwei30) [inline dubbo-common/src/test/java/com/alibaba/dubbo/common/logger/slf4j/Slf4jLoggerTest.java]: no '*'

File: dubbo-common/src/test/java/com/alibaba/dubbo/common/io/UnsafeByteArrayOutputStreamTest.java (hunk 3 above)
Comment (htynkn) [reply dubbo-common/src/test/java/com/alibaba/dubbo/common/io/UnsafeByteArrayOutputStreamTest.java]: @beiwei30 sorry, I thought checkstyle rule should check * import... but looks like it's only enable for source, not test folder.
I already fix * imports issue. Please refer to last commit

**Pull request merged**